Improve styling on the reverse dependencies page

dancristianb opened the pull request:
[WIP]

dancristianb commented:
This one is now ready for review. Let me know what you think, thanks! 🙏

simi commented:
Hello! Thanks for the PR!

Would it be possible to keep the content inside the layout? I tried to explain with the red arrow on image. See on gem page nothing is crossing the arrow which is inline with top grey border.

![image](https://user-images.githubusercontent.com/193936/235797863-e4a17830-c499-41cc-b659-8c7515bbcc65.png)

On reverse deps page, some content escapes the layout.

![image](https://user-images.githubusercontent.com/193936/235797810-b62f9083-b542-414e-9e71-dec43a73439b.png)

simi commented:
> Got a question here, should we maybe add a note something like "This gem has no reverse dependencies" just to be sure it's clear why we don't render anything? 

Yes, that makes sense to me.

dancristianb commented:
Thanks @simi for the comments! 🙏 I updated the reverse dependencies page layout so it now matches the gems page.

<img width="1237" alt="Screenshot 2023-05-04 at 22 04 15" src="https://user-images.githubusercontent.com/32739319/236304013-debd9d79-eced-45a2-88ec-842bfd2d67be.png">
<img width="1189" alt="Screenshot 2023-05-04 at 22 04 26" src="https://user-images.githubusercontent.com/32739319/236304025-049bbe65-8ae0-4863-959c-488fa4b9ee30.png">

And I've added in this `This gem has no reverse dependencies.` helper text right, hope that's OK.

<img width="1199" alt="Screenshot 2023-05-04 at 22 04 42" src="https://user-images.githubusercontent.com/32739319/236304033-9fbc0057-42bd-453b-9e6b-b24e7e00a0f8.png">

Let me know what you think, thanks!

simi commented:
Looks good. Thanks for the PR!
